uiowa/uiowa
# Emergency Website JSON Endpoint

Resolves #7550

Here is the output so far:

```json
// 20240503121038
// https://emergency.uiowa.ddev.site/api/active

{
  "jsonapi": {
    "version": "1.0",
    "meta": {
      "links": {
        "self": {
          "href": "http://jsonapi.org/format/1.0/"
        }
      }
    }
  },
  "data": [
    {
      "type": "node--hawk_alert",
      "id": "a76b6646-9dd7-4294-aa9f-52a941caa3c7",
      "links": {
        "self": {
          "href": "https://emergency.uiowa.ddev.site/api/active/a76b6646-9dd7-4294-aa9f-52a941caa3c7?resourceVersion=id%3A76"
        }
      },
      "attributes": {
        "alert": "HAWK ALERT NWS has issued a tornado warning for Johnson County until 4:45pm. Seek immediate shelter. See emergency.uiowa.edu for further information.",
        "date": "2024-04-16T21:21:37+00:00",
        "more_info_link": "https://emergency.uiowa.ddev.site",
        "complete": false
      }
    }
  ],
  "links": {
    "self": {
      "href": "https://emergency.uiowa.ddev.site/api/active"
    }
  }
}
```

# To Test

- `ddev blt ds --site=emergency.uiowa.edu`
- Go to https://emergency.uiowa.ddev.site/api/active in an anonymous browser window (it should be empty but working)
- Change the base url on [this line](https://github.com/uiowa/uiowa/blob/823fd555fd3a53330ed73deea699960b7d67025e/docroot/sites/emergency.uiowa.edu/modules/emergency_core/src/EmergencyAPI.php#L59) to `BASE_URL_1`
- Run `ddev drush @emergency.local rave-alerts` and see one item is imported.
- Refresh the anonymous session of https://emergency.uiowa.ddev.site/api/active and see the JSON output of the alert.
- Confirm that alert, date, more_info_link are included. The more_info_link will adjust to the environment it is running in and so in PROD it will eventually be https://emergency.uiowa.edu.
- `ddev drush @emergency.local uli` and go to the newly created alert and mark it as complete and save. Refresh the anonymous session of https://emergency.uiowa.ddev.site/api/active and see that the alert disappears.
- Confirm no other resource endpoints can be reached as they should be disabled.
